utils.py

```diff
@@ -69,7 +69,10 @@ def __str__(self):
 
 def txfDate(date):
```

Comment:
Maybe it's time we add type annotations to this code, it now takes `str | datetime` and returns `str`.